### DISCUSSION TITLE
Where is the payload in the "main" of a top-level statement Lambda function

### DISCUSSION THREAD

**Discussion by hounddog22030** [aws/aws-lambda-dotnet#1239] [General]
Where is the payload in the "main" of a top-level statement Lambda function?

Indirectly related to [this discussion](https://github.com/aws/aws-lambda-dotnet/discussions/1195):

Where is the payload in the main of a TLS Lambda function.

I have checked `args[]` but there is nothing there.

I would like to be able to read some of the payload before deciding how to setup my handler.  Specifically, I need to get the S3 Arn of an assembly and bring it down.  In my function, I have the S3 arn as element of the JSON payload (an Lambda function called by a Cloudformation::CustomResource).

I would like to bring that assembly down before setting up my handler or bootstrapping.

Thanks!

**Comment by ashishdhingra** [comment]
@hounddog22030 Good afternoon. I'm unsure what you meant by payload in `main` of a top-level statement Lambda function. Could you please elaborate with an example, may be comparing on how you would do with a normal class library Lambda function that doesn't use top-level statements.

Please refer `Executable assemblies` section in blog post https://aws.amazon.com/blogs/compute/introducing-the-net-6-runtime-for-aws-lambda/ where it specifies that:
- With top-level statements, the Main() method has been generated by the compiler.
- When using top-level statements, you instead tell Lambda to run the assembly, which runs the top-level statements. To indicate that you want Lambda to run the assembly, set the Lambda function handler to the assembly name only.
- With the .NET assembly containing the Lambda function being run at startup, instead of the Lambda runtime client, your function code must start the Lambda runtime client so that Lambda events are sent to your code.
  ```C#
  await LambdaBootstrapBuilder.Create(handler, new DefaultLambdaJsonSerializer())
        .Build()
        .RunAsync();
  ```

what @normj is suggesting in [discussion](https://github.com/aws/aws-lambda-dotnet/discussions/1195#discussioncomment-2961622) is that you could define your own environment variable and then check that environment variable to decide which handler to register with the Lambda bootstrap. We are not sure if it fits your use case though.

Thanks,
Ashish
